geopyspark/geotrellis/catalog.py

```diff
@@ -577,18 +576,19 @@ def write(geopysc,
 
     builds = _mapped_builds[uri]
 
-    key = geopysc.map_key_input(rdd_type, True)
-
-    schema = geopysc.create_schema(key)
-
     if not time_unit:
         time_unit = ""
 
-    builds.writer.write(key,
-                        layer_name,
-                        layer_zoom,
-                        rdd._jrdd,
-                        schema,
-                        json.dumps(metadata),
-                        time_unit,
-                        index_strategy)
+    if isinstance(tiled_raster_rdd, TiledRasterRDD):
```

Comment:
@echeipesh I changed this code in a later Pr: https://github.com/jbouffard/geopyspark/blob/pyramid-perf/geopyspark/geotrellis/catalog.py#L582